framework/framework.go

```diff
@@ -173,22 +132,21 @@ func (f *Framework) AddFunc(obj interface{}) {
 	f.mutex.Lock()
 	defer f.mutex.Unlock()
 
-	ctx := context.Background()
-	ctx, err := f.initCtxFunc(ctx, obj)
+	resourceSet, err := f.resourceRouter.ResourceSet(obj)
 	if err != nil {
-		f.logger.LogCtx(ctx, "error", fmt.Sprintf("%#v", err), "event", "create")
+		f.logger.Log("error", fmt.Sprintf("%#v", err), "event", "create")
 		return
 	}
 
-	rs, err := f.resourceRouter(ctx, obj)
+	ctx, err := resourceSet.InitCtx(context.Background(), obj)
 	if err != nil {
-		f.logger.LogCtx(ctx, "error", fmt.Sprintf("%#v", err), "event", "create")
+		f.logger.Log("error", fmt.Sprintf("%#v", err), "event", "create")
```

Comment:
Why not LogCtx? It provides a very useful information, like object name.

Reply:
The context was tried to be created but an error was returned. There should not be any value in the context since a convention on error is to return the zero value. Below makes sense. Here not IMO.

Reply:
+1 this if doesn't make sense. I meant in general. I thing it'd be good to still use that below.

Reply:
When you check the code it is sorted below.